ilastik/utility/exportFile.py

```diff
@@ -447,17 +447,25 @@ def write_all(self, mode, compression=None):
         count = 0
         self.ExportProgress(0)
         if mode in ("h5", "hd5", "hdf5"):
-            with h5py.File(self.file_name, "w") as fout:
-                for table_name, table in self.table_dict.items():
-                    self._make_h5_dataset(
-                        fout,
-                        table_name,
-                        table,
-                        self.meta_dict.get(table_name, {}),
-                        compression if compression is not None else {},
-                    )
-                    count += 1
-                    self.ExportProgress(count * 100 / len(self.table_dict))
+            for libver in ["earliest", "v108"]:
```

Comment:
in principle one could try all values in [`h5py._hl.files.libver_dict`](https://github.com/h5py/h5py/blob/8bed19bed35108190e598993d26bacb70879059c/h5py/_hl/files.py#L35) - do you know of any changes that occurred after `v108` that address the issue of object headers?

Otherwise I would prefer to check first if any of the tables can be written beforehand, but it seems not obvious to determine header size from a structured array.